src/Domain.LinnApps/Reports/PurchaseOrdersReportService.cs

```diff
@@ -163,18 +170,97 @@ public ResultsModel GetOrdersByPartReport(DateTime from, DateTime to, string par
+                                            RowId = rowId,
+                                            ColumnId = "Supplier Id",
+                                            TextDisplay = supplier.SupplierId.ToString()
+                                        });
+                supplierResults.Add(new CalculationValueModel { RowId = rowId, ColumnId = "Supplier Name", TextDisplay = supplier.SupplierName });
+                supplierResults.Add(new CalculationValueModel { RowId = rowId, ColumnId = "view", TextDisplay = "view" });
+                supplierResults.Add(new CalculationValueModel { RowId = rowId, ColumnId = "csv", TextDisplay = "csv" });
+            }
+
+            results.ValueDrillDownTemplates.Add(new DrillDownModel("view", "/purchasing/reports/unacknowledged-orders?supplierId={rowId}", null, 2));
+            results.ValueDrillDownTemplates.Add(new DrillDownModel("csv", "/purchasing/reports/unacknowledged-orders/export?supplierId={rowId}", null, 3));
+            this.reportingHelper.AddResultsToModel(results, supplierResults, CalculationValueModelType.TextValue, true);
+            this.SortRowsByTextColumnValues(results, 1);
+
+            return results;
+        }
+
+        public void SortRowsByTextColumnValues(
+            ResultsModel model,
+            int columnIndex,
+            int? column2Index = null,
+            int? column3Index = null,
+            bool sortDescending = false)
+        {
+            var sortOrder = 0;
+
+            if (sortDescending)
+            {
+                foreach (var modelRow in model.Rows
+                    .OrderByDescending(a => model.GetGridTextValue(a.RowIndex, columnIndex))
+                    .ThenByDescending(t => column2Index.HasValue ? model.GetGridTextValue(t.RowIndex, column2Index.Value) : string.Empty)
+                    .ThenByDescending(t => column3Index.HasValue ? model.GetGridTextValue(t.RowIndex, column3Index.Value) : string.Empty)
+                    .ThenBy(a => a.RowTitle))
+                {
+                    modelRow.SortOrder = sortOrder++;
+                }
+            }
+            else
+            {
+                foreach (var modelRow in model.Rows
+                    .OrderBy(a => model.GetGridTextValue(a.RowIndex, columnIndex))
+                    .ThenBy(a => column2Index.HasValue ? model.GetGridTextValue(a.RowIndex, column2Index.Value) : string.Empty)
+                    .ThenBy(a => column3Index.HasValue ? model.GetGridTextValue(a.RowIndex, column3Index.Value) : string.Empty)
+                    .ThenBy(a => a.RowTitle))
+                {
+                    modelRow.SortOrder = sortOrder++;
+                }
+            }
+        }
+
+        public ResultsModel GetUnacknowledgedOrders(DateTime startDate, DateTime endDate, int supplierId, int? organisationId)
+        {
+            throw new NotImplementedException();
+        }
+
         private void AddSupplierReportColumns(SimpleGridLayout reportLayout)
```

Comment:
I said it on Iain's PR so I'll say it here for equality: these could be static. I don't know if we care really but apparently its more performant when methods that can be static are static

Reply:
I think this is one of the methods I've just made static so will just need merge conflicts resolved. Might need to check the files you've moved too, I think I made a change or two to PurchaseOrder.cs, PurchaseOrderDelivery and PurchaseOrderDetails and it doesn't look like it'll catch it as a merge conflict because I've made the changes to them in their old location